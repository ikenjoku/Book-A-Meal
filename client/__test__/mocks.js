
export const meals = [
  {
    id: 0,
    name: 'Nigerian Jollof',
    description: 'Max Jollof rice, plantain, grilled turkey, bellefull Coke',
    price: 1100,
    imageurl: 'https://africa-publicmages/products/80/80418_1465475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 1,
    name: 'Rice and Beans',
    description: 'Rice, Beans, Plantain, Panla Sauce, Max Coke',
    price: 1500,
    imageurl: 'https://africa-g/images/products/80/80418_1465475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 2,
    name: 'Egusi Semo',
    description: 'Semolina, Egusi Soup, Meat, Extra',
    price: 1200,
    imageurl: 'https://africa-public.food.jcts/80/80418_1465475724_ma.jpg',
  },
  {
    id: 3,
    name: 'Green Calulu',
    description: 'Chicken salad with extra cabbage',
    price: 1250,
    imageurl: 'https://dynamic/production/ng/images/products/80/_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 4,
    name: 'Onion Jappatti',
    description: 'Ingera palate with raw meat sauce',
    price: 2200,
    imageurl: 'https://dynamic/production/ng/images/produc475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
];

export const testMeal = {
  id: 3234,
  name: 'Strawberry flakes',
  description: 'Ingera palate with raw meat sauce',
  price: 2200,
  imageurl: 'https://dynamic/prodg/images/produc475724_ma.jpg',
  createdAt: new Date(),
  updatedAt: new Date(),
};

export const mockStoreData = {
  authReducer: {
    authLoading: true,
    user: {},
    isLoggedIn: false,
    error: null,
  },
  mealReducer: {
    meals: [
      {
        id: 3,
        name: 'Green Calulu',
        description: 'Chicken salad with extra cabbage',
        price: 1250,
        imageurl: 'https://dynamic/production/ng/images/products/80/_ma.jpg',
        createdAt: new Date(),
        updatedAt: new Date(),
      },
      {
        id: 4,
        name: 'Onion Jappatti',
        description: 'Ingera palate with raw meat sauce',
        price: 2200,
        imageurl: 'https://dynamic/production/ng/images/produc475724_ma.jpg',
        createdAt: new Date(),
        updatedAt: new Date(),
      },
    ],
    error: null,
    isLoading: false,

  },
  menuReducer: {
    selectedMenu: null,
    error: null,

  },
  setMenuReducer: {
    createdMenu: null,
    selectedDate: '',
    error: null,
    success: false,
    updatedMenu: null,
  },
  orderReducer: {
    orderedMeal: null,
    error: null,
    meals: [],
    orders: [],
    previousOrders: [],
    isLoadingOrders: false,
  },
};

export const createdMenu = {
  message: 'Menu for 2018-06-15 was successfully created',
  newMenu: {
    id: 24,
    date: '2018-06-15',
    updatedAt: '2018-07-02T17:37:46.023Z',
    createdAt: '2018-07-02T17:37:46.023Z',
  },
};

export const menus = [{
  id: 23,
  date: '2018-07-02',
  createdAt: '2018-07-01T23:56:23.508Z',
  updatedAt: '2018-07-01T23:56:23.508Z',
  Meals: [
    {
      id: 31,
      name: 'Beans and Bread',
      price: 800,
      description: 'Heavily loaded with smoked fish.',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529133645/bookameal/2018-06-16T07:20:43.822ZVegetablerice%2Bplantain%2Bgoatmeat%2Bcoke.jpg.jpg',
      MealMenu: {
        mealId: 31,
        menuId: 23,
        createdAt: '2018-07-01T23:56:28.012Z',
        updatedAt: '2018-07-01T23:56:28.012Z',
      },
    },
    {
      id: 34,
      name: 'Fish balls soup',
      price: 1200,
      description: 'Deliciously light soup with white fish balls',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529134583/bookameal/2018-06-16T07:36:22.300Zassortedmeat.jpg.jpg',
      MealMenu: {
        mealId: 34,
        menuId: 23,
        createdAt: '2018-07-01T23:56:32.313Z',
        updatedAt: '2018-07-01T23:56:32.313Z',
      },
    },
    {
      id: 30,
      name: 'Beef with veggie soup',
      price: 1150,
      description: 'Stir fried sliced beef with premium vegetables and spices',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529136037/bookameal/2018-06-16T08:00:34.302ZUgwu-Archi.jpg.jpg',
      MealMenu: {
        mealId: 30,
        menuId: 23,
        createdAt: '2018-07-01T23:56:36.920Z',
        updatedAt: '2018-07-01T23:56:36.920Z',
      },
    },
  ],
},
{
  id: 24,
  date: new Date().toISOString().substr(0, 10),
  createdAt: '2018-07-01T23:56:23.508Z',
  updatedAt: '2018-07-01T23:56:23.508Z',
  Meals: [
    {
      id: 31,
      name: 'Beans and Bread',
      price: 800,
      description: 'Heavily loaded with smoked fish.',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529133645/bookameal/2018-06-16T07:20:43.822ZVegetablerice%2Bplantain%2Bgoatmeat%2Bcoke.jpg.jpg',
      MealMenu: {
        mealId: 31,
        menuId: 23,
        createdAt: '2018-07-01T23:56:28.012Z',
        updatedAt: '2018-07-01T23:56:28.012Z',
      },
    },
    {
      id: 34,
      name: 'Fish balls soup',
      price: 1200,
      description: 'Deliciously light soup with white fish balls',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529134583/bookameal/2018-06-16T07:36:22.300Zassortedmeat.jpg.jpg',
      MealMenu: {
        mealId: 34,
        menuId: 23,
        createdAt: '2018-07-01T23:56:32.313Z',
        updatedAt: '2018-07-01T23:56:32.313Z',
      },
    },
    {
      id: 30,
      name: 'Beef with veggie soup',
      price: 1150,
      description: 'Stir fried sliced beef with premium vegetables and spices',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529136037/bookameal/2018-06-16T08:00:34.302ZUgwu-Archi.jpg.jpg',
      MealMenu: {
        mealId: 30,
        menuId: 23,
        createdAt: '2018-07-01T23:56:36.920Z',
        updatedAt: '2018-07-01T23:56:36.920Z',
      },
    },
  ],
}];

export const orders = [
  {
    id: 2,
    date: '2018-05-16',
    amount: 1500,
    userId: 1,
    mealId: 2,
    status: 'cancelled',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 1,
      firstname: 'Ezinne',
      lastname: 'Njoku',
    },
    Meal: {
      id: 2,
      name: 'Rice and Beans',
      price: 1500,
    },
  },
  {
    id: 3,
    date: '2018-05-16',
    amount: 1200,
    userId: 2,
    mealId: 3,
    status: 'pending',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 2,
      firstname: 'Kelechi',
      lastname: 'Njoku',
    },
    Meal: {
      id: 3,
      name: 'Egusi Semo',
      price: 1200,
    },
  },
  {
    id: 4,
    date: '2018-05-16',
    amount: 1500,
    userId: 3,
    mealId: 3,
    status: 'delivered',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 3,
      firstname: 'Nnamdi',
      lastname: 'James',
    },
    Meal: {
      id: 3,
      name: 'Egusi Semo',
      price: 1200,
    },
  },
];
