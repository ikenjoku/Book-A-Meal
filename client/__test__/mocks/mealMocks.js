export const getAllMealsMock = [
  {
    name: 'test meal',
    price: 1230,
    description: 'this is a very nice meal',
    imageurl: 'https://sdfsdjhbsn/sdf.asp',
  },
];
