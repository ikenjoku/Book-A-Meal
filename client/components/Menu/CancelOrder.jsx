import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { cancelAnOrder } from '../../actions/orderActions';

export const CancelOrder = ({ orderId, closeModal, cancelAnOrder }) =>
  (
    <div className="confirm-modal">
      <h3>Are you sure you want to cancel this order?</h3>
      <div>
        <button
          id="confirm-cancel-order"
          onClick={() => {
          cancelAnOrder(orderId);
          closeModal();
          }}
        >Yes
        </button>
        <button onClick={closeModal}>No</button>
      </div>
    </div>
  );

CancelOrder.propTypes = {
  orderId: PropTypes.number.isRequired,
  closeModal: PropTypes.func.isRequired,
  cancelAnOrder: PropTypes.func.isRequired,
};

export default connect(undefined, { cancelAnOrder })(CancelOrder);
