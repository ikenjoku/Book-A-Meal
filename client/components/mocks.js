

export const sampleOrders = {
  message: 'All orders retrieved successfully',
  orders: [
    {
      id: 2,
      date: '2018-05-16',
      amount: 1500,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 3,
      date: '2018-05-16',
      amount: 1200,
      userId: 2,
      mealId: 3,
      status: 'pending',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 4,
      date: '2018-05-16',
      amount: 1500,
      userId: 3,
      mealId: 3,
      status: 'delivered',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 3,
        firstname: 'Nnamdi',
        lastname: 'James',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 1,
      date: '2018-05-16',
      amount: 1100,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T20:38:56.387Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 5,
      date: '2018-05-23',
      amount: 1500,
      userId: 2,
      mealId: 2,
      status: 'pending',
      createdAt: '2018-05-23T23:55:05.091Z',
      updatedAt: '2018-05-23T23:55:05.091Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 7,
      date: '2018-05-16',
      amount: 1200,
      userId: 2,
      mealId: 3,
      status: 'pending',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 6,
      date: '2018-05-16',
      amount: 1500,
      userId: 3,
      mealId: 3,
      status: 'delivered',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 3,
        firstname: 'Nnamdi',
        lastname: 'James',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 8,
      date: '2018-05-16',
      amount: 1100,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T20:38:56.387Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
  ],
};

