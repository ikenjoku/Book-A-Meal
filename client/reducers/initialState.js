export default {
  authReducer: {
    authLoading: true,
    user: {},
    isLoggedIn: false,
    error: null,
  },
  mealReducer: {
    meals: [],
    error: null,
    isLoading: true,

  },
  menuReducer: {
    selectedMenu: null,
    error: null,

  },
  setMenuReducer: {
    createdMenu: null,
    selectedDate: '',
    error: null,
    success: false,
    updatedMenu: null,
  },
  orderReducer: {
    orderedMeal: null,
    error: null,
    meals: [],
    orders: [],
    previousOrders: [],
    isLoadingOrders:false,
  },
};
