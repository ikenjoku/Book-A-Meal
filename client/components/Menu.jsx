import React, { Component } from 'react';
import PropTypes from 'prop-types';

import NavBar from './NavBar.jsx';

import { menuSample } from "./mocks";

export default class Menu extends Component {
  constructor(props){
    super(props)
  }

  render(){
    return (
    <div>
      <NavBar/>
      <main className="menu-list-content">
        <div className="menu-list">
          <MenuTable/>
        </div>
        <div className="order-summary">
          <div className="order-summary-wrap">
            <PreviousOrderTable/>
            <CurrentOrderTable/>
          </div>
        </div>
      </main>
      <footer className="footer">
          <p>Book-A-Meal &copy;2018</p>
      </footer>     
    </div>
    );
  }
}

class MenuTable extends Component{
  constructor(props){
    super(props);
    this.state = {
      meals: menuSample.Meals
    }
  }

  render(){
    return (
      <div>
      <h2 className="center cool-lg-text">Menu List</h2>
        <MenuSearch/>
        <MealList meals={this.state.meals}/>
      </div>
    );
  }
}

class MenuSearch extends Component{
  constructor(props){
    super(props);
  }

  render(){
    return (
      <div>
        <div className="date-picker-pad order-history-date-picker center">
          <form>
            <input 
              id="datePicker" 
              className="input-field-control" 
              type="date" 
              placeholder="Choose a date..." 
              name="date"/>
            <input 
              id="datePickerBtn" 
              className="input-btn-control" 
              type="submit" 
              value="Choose date"/>
          </form>
        </div>
      </div>
    );
  }
}

const MealList = ({meals}) => {

  return (
    <div>
      {meals.map(meal =>
        <div key={meal.id} className="food-item">
          <img src={ require("../static/mealImg.jpg") } alt="" />
          <div className="food-title-desc">
              <p className="food-item-title">{ meal.name }</p>
              <p className="food-item-detail">{ meal.description }</p>
          </div>
          <div className="food-item-price">
              { meal.price }
          </div>
          <div><button className="food-add-btn">Order</button></div>
      </div> 
      )}
    </div>
  );
}

const PreviousOrderTable = (props) => {

  return (
    <div>
    <h4>Previous orders</h4>
    <div className="order-item">
        <div className="order-item-qty">07/3/17</div>
        <div className="order-item-name">Utazi</div>
        <div className="order-item-price">N1400 </div>
    </div>
    <div className="order-item">
        <div className="order-item-qty">12/4/17</div>
        <div className="order-item-name">Jollof rice</div>
        <div className="order-item-price">N200 </div>
    </div>
    <div className="order-item">
        <div className="order-item-qty">15/3/18</div>
        <div className="order-item-name">Boiled Egg</div>
        <div className="order-item-price">N100 </div>
    </div>
    </div>
  );
}


const CurrentOrderTable = (props) => {
  return (
    <div>
    <h4>Current Order</h4>
      <div className="order-item">
        <div className="order-item-qty">Egusi Soup</div>
        <div className="order-item-name">N1200</div>
        <div className="order-item-price"><button className="cancel-order-btn">Cancel Order</button> </div>
      </div>
      <hr/>
    </div>
  );
}
