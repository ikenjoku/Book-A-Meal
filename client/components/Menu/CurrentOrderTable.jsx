import React, { Component } from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import PropTypes from 'prop-types';
import modalStyle from '../../utils/modalStyle';
import CancelOrder from './CancelOrder';
import UpdateOrder from './UpdateOrder';
import { getAllPreviousOrders } from '../../actions/orderActions';

export class CurrentOrderTable extends Component {
  constructor(props) {
    super(props);
    this.state = {
      openCancelModal: false,
      openUpdateModal: false,
      orderId: undefined,
      orderToUpdate: undefined,
    };
  }

  componentDidMount() {
    this.props.getAllPreviousOrders(this.props.userId);
  }

  openCancelModal = ({ orderId }) => {
    this.setState(() => ({ orderId, openCancelModal: true }));
  }

  closeCancelModal = () => {
    this.setState(() => ({ openCancelModal: false }));
  }

  handleOrderUpdate = (orderToUpdate) => {
    this.setState(() => ({ orderToUpdate, openUpdateModal: true }));
  }

  closeUpdateModal = () => {
    this.setState(() => ({ openUpdateModal: false }));
  }

  render() {
    const { currentOrders } = this.props;
    const pendingOrders = currentOrders.filter(order => order.status === 'pending');

    return (
      <div>
        <h4>Current Order</h4>
        {pendingOrders.length === 0 ? <p>No current orders</p> :
          pendingOrders.map(order =>
            (
              <div key={order.id} className="order-item">
                <div className="order-item-name">{order.Meal.name}</div>
                <div className="order-item-amount">&#8358; {order.amount}</div>
                <div className="order-item-actions">
                  <button
                    className="update-order-icon"
                    onClick={() => { this.handleOrderUpdate(order); }}
                  >
                    <i className="far fa-edit" />
                  </button>
                  <button
                    className="cancel-order-icon"
                    onClick={() => {
                      this.openCancelModal({ orderId: order.id });
                    }}
                  >
                    <i className="far fa-times-circle 5x" />
                  </button>
                </div>
              </div>
            ))
          }
        <hr />
        <Modal
          isOpen={this.state.openCancelModal}
          contentLabel="cancel-order"
          style={modalStyle}
          ariaHideApp={false}
        >
          <div className="close-icon">
            <button
              onClick={this.closeCancelModal}
            >
              <i className="fas fa-times fa-2x" />
            </button>
          </div>
          <CancelOrder
            closeModal={this.closeCancelModal}
            orderId={this.state.orderId}
          />
        </Modal>
        <Modal
          isOpen={this.state.openUpdateModal}
          contentLabel="update-order"
          style={modalStyle}
          ariaHideApp={false}
        >
          <div className="close-icon">
            <button
              onClick={this.closeUpdateModal}
            >
              <i className="fas fa-times fa-2x" />
            </button>
          </div>
          <UpdateOrder
            closeModal={this.closeUpdateModal}
            orderToUpdate={this.state.orderToUpdate}
          />
        </Modal>
      </div>
    );
  }
}

CurrentOrderTable.propTypes = {
  currentOrders: PropTypes.arrayOf(PropTypes.object).isRequired,
  userId: PropTypes.number.isRequired,
  getAllPreviousOrders: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  currentOrders: state.orderReducer.previousOrders,
  userId: state.authReducer.user.id,
});

export default connect(mapStateToProps, { getAllPreviousOrders })(CurrentOrderTable);
