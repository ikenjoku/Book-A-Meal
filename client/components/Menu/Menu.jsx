import React from 'react';
import MenuTable from './MenuTable';
import PreviousOrderTable from './PreviousOrderTable';
import CurrentOrderTable from './CurrentOrderTable';

const Menu = () =>
  (
    <main id="menu-page" className="menu-list-content">
      <div className="menu-list">
        <MenuTable />
      </div>
      <div className="order-summary">
        <div className="order-summary-wrap">
          <CurrentOrderTable />
          <PreviousOrderTable />
        </div>
      </div>
    </main>
  );

export default Menu;

