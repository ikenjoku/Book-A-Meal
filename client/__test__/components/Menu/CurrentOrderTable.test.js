import React from 'react';
import { shallow } from "enzyme";
import { CurrentOrderTable, mapStateToProps } from "../../../components/Menu/CurrentOrderTable";
import initialState from '../../../reducers/initialState'
import { orders } from '../../mocks';

describe('Menu: CurrentOrderTable Component', () => {
  let currentOrders, wrapper, userId, getAllPreviousOrders;
  currentOrders = orders;
  userId = 1;
  getAllPreviousOrders = jest.fn();
  test('Should render CurrentOrderTable correctly', () => {
    wrapper = shallow(
      <CurrentOrderTable
        currentOrders={currentOrders}
        userId={userId}
        getAllPreviousOrders={getAllPreviousOrders}
      />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render a message when there are no pending orders', () => {
    wrapper = shallow(
      <CurrentOrderTable
        currentOrders={[orders[0], orders[2]]}
        userId={userId}
        getAllPreviousOrders={getAllPreviousOrders}
      />);
    expect(wrapper.find('.order-item').length).toBe(0);
  });

  test('should handleOrderUpdate  when update button is clicked', () => {
    wrapper = shallow( <CurrentOrderTable 
      currentOrders={currentOrders} 
      userId={userId}
      getAllPreviousOrders={getAllPreviousOrders}
      />);

    const handleOrderUpdateSpy = jest.spyOn(
      wrapper.instance(), 'handleOrderUpdate'
    );
    const orderToUpdate = orders[1]
    wrapper.instance().handleOrderUpdate(orderToUpdate);
    expect(handleOrderUpdateSpy).toHaveBeenCalledTimes(1);
  });

  test('should close update modal when No button is clicked', () => {
    wrapper = shallow( <CurrentOrderTable 
      currentOrders={currentOrders} 
      userId={userId}
      getAllPreviousOrders={getAllPreviousOrders}
      />);

    const closeUpdateModalSpy = jest.spyOn(
      wrapper.instance(), 'closeUpdateModal'
    );
    wrapper.instance().closeUpdateModal();
    expect(closeUpdateModalSpy).toHaveBeenCalledTimes(1);
  });

  test('should openCancelModal when cancel button is clicked', () => {
    wrapper = shallow( <CurrentOrderTable 
      currentOrders={currentOrders} 
      userId={userId}
      getAllPreviousOrders={getAllPreviousOrders}
      />);

    const openCancelModalSpy = jest.spyOn(
      wrapper.instance(), 'openCancelModal'
    );
    wrapper.instance().openCancelModal({ orderId: 2 });
    expect(openCancelModalSpy).toHaveBeenCalledTimes(1);
  });

  test('should close order-cancel modal when No button is clicked', () => {
    wrapper = shallow(<CurrentOrderTable 
      currentOrders={currentOrders} 
      userId={userId}
      getAllPreviousOrders={getAllPreviousOrders}
      />);

    const closeCancelModalSpy = jest.spyOn(
      wrapper.instance(), 'closeCancelModal'
    );
    wrapper.instance().closeCancelModal();
    expect(closeCancelModalSpy).toHaveBeenCalledTimes(1);
  });

  test('should call handleOrderUpdate when Edit button is clicked', () => {
    wrapper = shallow(<CurrentOrderTable 
      currentOrders={currentOrders} 
      userId={userId}
      getAllPreviousOrders={getAllPreviousOrders}
      />);

    const handleOrderUpdateSpy = jest.spyOn(
      wrapper.instance(), 'handleOrderUpdate'
    );
    wrapper.find('.update-order-icon').simulate('click', orders[0]);
    expect(handleOrderUpdateSpy).toHaveBeenCalledTimes(1);
  });

  test('should call openCancelModal when Cancel button is clicked', () => {
    wrapper = shallow(<CurrentOrderTable 
      currentOrders={currentOrders} 
      userId={userId}
      getAllPreviousOrders={getAllPreviousOrders}
      />);

    const openCancelModalSpy = jest.spyOn(
      wrapper.instance(), 'openCancelModal'
    );
    wrapper.find('.cancel-order-icon').simulate('click', {orderId: 2});
    expect(openCancelModalSpy).toHaveBeenCalledTimes(1);
  });

  test('should map state to props', () => {
    const ownProps = { currentOrders: orders };
    const tree = mapStateToProps(initialState, ownProps);
    expect(tree).toMatchSnapshot();
  });
});
