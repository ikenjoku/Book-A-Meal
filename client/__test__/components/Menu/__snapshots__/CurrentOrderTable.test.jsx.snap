// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu: CurrentOrderTable Component Should render CurrentOrderTable correctly 1`] = `
<div>
  <h4>
    Current Order
  </h4>
  <div
    className="order-item"
    key="3"
  >
    <div
      className="order-item-name"
    >
      Egusi Semo
    </div>
    <div
      className="order-item-amount"
    >
      â‚¦ 
      1200
    </div>
    <div
      className="order-item-actions"
    >
      <button
        className="update-order-icon"
        onClick={[Function]}
      >
        <i
          className="far fa-edit"
        />
      </button>
      <button
        className="cancel-order-icon"
        onClick={[Function]}
      >
        <i
          className="far fa-times-circle 5x"
        />
      </button>
    </div>
  </div>
  <hr />
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentLabel="cancel-order"
    isOpen={false}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "bottom": "auto",
          "left": "50%",
          "marginRight": "-50%",
          "minWidth": "20rem",
          "overflow": "hidden",
          "right": "auto",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "40%",
        },
        "overlay": Object {
          "backgroundColor": "rgba(0, 0, 0, 0.25)",
        },
      }
    }
  >
    <div
      className="close-icon"
    >
      <button
        onClick={[Function]}
      >
        <i
          className="fas fa-times fa-2x"
        />
      </button>
    </div>
    <Connect(CancelOrder)
      closeModal={[Function]}
    />
  </Modal>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentLabel="update-order"
    isOpen={false}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "bottom": "auto",
          "left": "50%",
          "marginRight": "-50%",
          "minWidth": "20rem",
          "overflow": "hidden",
          "right": "auto",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "40%",
        },
        "overlay": Object {
          "backgroundColor": "rgba(0, 0, 0, 0.25)",
        },
      }
    }
  >
    <div
      className="close-icon"
    >
      <button
        onClick={[Function]}
      >
        <i
          className="fas fa-times fa-2x"
        />
      </button>
    </div>
    <Connect(UpdateOrder)
      closeModal={[Function]}
    />
  </Modal>
</div>
`;

exports[`Menu: CurrentOrderTable Component should map state to props 1`] = `
Object {
  "currentOrders": Array [],
  "userId": undefined,
}
`;
